### PR TITLE
Module Options - Add apps.py to Django modules and fix `get_options`

### PR DESCRIPTION
**Pull request by danielsousaio**
## Type of PR

- [x] Bugfix
- [x] Minor changes

Relates to https://github.com/crowdbotics/django-scaffold/pull/396

## Changes introduced

Changes introduced:
- Add `apps.py` on all Django modules so that our `manifest.py` loading mechanism works
- Fix `get_options` call, BE code generation uses full module name for key
- Update modules json via `yarn run bootstrap` 
- Rename all django modules to use underscore as word separator (both in module path and apps.py)

## Test and review

Checkout and then run:
```sh
yarn run demo
yarn run add django-articles
cd demo/backend

pipenv install --dev
cp .env.template .env # and modify it, remove DATABASE_URL line to use sqlite

pipenv shell
python manage.py migrate # should do modules.articles migrations
python manage.py show_urls # should boot up django correctly and show urls
python manage.py shell # manual testing if you want
# > from django.conf import settings
# > settings.INSTALLED_APPS
# > => [...modules.articles]
```

### CODE REVIEW

File: modules/django-social-auth/apps.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from django.apps import AppConfig
+
+
+class SocialAuthConfig(AppConfig):
+    name = "modules.social-auth"
```
[the remark this answers is not in the document]
Comment (danielsousaio) [reply modules/django-social-auth/apps.py:5]: It's actually a valid python path:
```shell
yarn run demo
yarn run add django-social-auth
```
```shell
python manage.py show_urls | grep "modules."
# => /modules/social-auth/apple/connect/     modules.social-auth.views.AppleLogin    social_apple_connect
# => /modules/social-auth/apple/login/       modules.social-auth.views.AppleLogin    social_apple_login
# => /modules/social-auth/facebook/connect/  modules.social-auth.views.FacebookLogin social_facebook_connect
# => /modules/social-auth/facebook/login/    modules.social-auth.views.FacebookLogin social_facebook_login
# => /modules/social-auth/google/connect/    modules.social-auth.views.GoogleLogin   social_google_connect
# => /modules/social-auth/google/login/      modules.social-auth.views.GoogleLogin   social_google_login
# => /modules/social-auth/socialaccounts/    rest_auth.registration.views.SocialAccountListView      social_account_list
# => /modules/social-auth/socialaccounts/<int:pk>/disconnect/        rest_auth.registration.views.SocialAccountDisconnectView      social_account_disconnect
```

The reason why I named all `apps.py` with `-` instead of `_` was because we are adding modules dynamically into `INSTALLED_APPS`, and the name of the modules is derived from the module directory path:
https://github.com/crowdbotics/django-scaffold/blob/ecff5e4334d864e9d21db89e2eeae9b109dc39f9/%7B%7Bcookiecutter.project_slug%7D%7D/%7B%7Bcookiecutter.project_slug%7D%7D/settings.py#L91
https://github.com/crowdbotics/django-scaffold/blob/ecff5e4334d864e9d21db89e2eeae9b109dc39f9/%7B%7Bcookiecutter.project_slug%7D%7D/modules/manifest.py#L14-L15

In other words, naming all `AppConfig.name` with dashes was intentional, so that it follows the registered name in `INSTALLED_APPS`.

[the remark this answers is not in the document]
Comment (math-a3k) [reply modules/django-social-auth/apps.py:5]: To my surprise it is legal, but you have to resort to this to use it from others part of the code:
https://stackoverflow.com/a/37831973

[the remark this answers is not in the document]
Comment (danielsousaio) [reply modules/django-social-auth/apps.py]: I've updated all django modules to use underscores

[the remark this answers is not in the document]
Comment (math-a3k) [reply modules/django-social-auth/apps.py]: @danielsousaio it's good, but why don't we focus on PLAT-7015? It has all the refactor to python modules?

File: modules/django-articles/articles/models.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from django.db import models
 from modules.utils import get_options
 
-MEDIA_UPLOAD_PATH = get_options("django-articles", "MEDIA_UPLOAD_PATH")
+MEDIA_UPLOAD_PATH = get_options("django_articles", "MEDIA_UPLOAD_PATH")
```
Comment (math-a3k) [inline modules/django-articles/articles/models.py:5]: The other was OK, here it should be the module slug because is how options are generated in the backend, see https://github.com/crowdbotics/modules/blob/modules_as_pkgs/docs/authoring.md#the-python-package-content

**Review by math-a3k — CHANGES_REQUESTED** [review]

File: modules/django-articles/articles/models.py (hunk 2 above)
Comment (danielsousaio) [reply modules/django-articles/articles/models.py:5]: Okay let me rebase and drop it, thanks.

**Comment by math-a3k** [comment]
@danielsousaio there is a bug related to options:
https://github.com/crowdbotics/django-scaffold/commit/683c8554146338c65c2322ddfb6fb91b08656ea0
This means that in order to use the options set by the user, the directories of the apps should be the slug snake-cased
